algorithms/arrays/garage.py

```diff
@@ -65,4 +150,64 @@ def garage(initial, final):
 2 3 0 1 4 -- zero = 2, t, car_to_move = final[2] = 2, 
              pos = initial.index(2) = 0, switched [0], [2]
 0 3 2 1 4 -- initial == final
-"""
\ No newline at end of file
+"""
+
+'''
+This function doesn't compute a sequence of steps. It only computes
+the minimum number of steps to rearrange. But this function is much
+faster.
+
+Time complexity O(n)
+Space complexity O(n)
+'''
+def garage_num(initial, final, simulation=False):
+    n = len(initial)
+    assert n == len(final)
+
+    d, e, f = {}, {}, {}
+    for i in range(n):
+        d[ initial[i] ] = i
+        e[i] = initial[i]
+        f[ final[i] ] = i
+
+    def move_car_to_correct_spot(car):
+        curr_pos_of_car = d[car]
+        pos_car_shld_be_in = f[car]
+        car_in_its_spot = e[pos_car_shld_be_in]
+
+        d[car_in_its_spot] = curr_pos_of_car
+        d[car] = pos_car_shld_be_in
+
+        e[curr_pos_of_car] = car_in_its_spot
+        e[pos_car_shld_be_in] = car
+        return
+
+    move_car_to_correct_spot(0)
+    j = 1
+    for car in d:
+        if d[car] != f[car]:
+            move_car_to_correct_spot(car)
+            j += 3
+
+    if simulation: assert d == f
+    return j
+
+def test(num, initial, final, min_steps):
```

Comment:
Thanks for adding tests. But please move the tests to `tests/`, see other test suites for details. And don't execute the functions in this file (and no `main`).